// @qdrant/js-client-rest/dist/cjs/api-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createClient = exports.createApis = void 0;
const cluster_api_js_1 = require("./api/cluster-api.js");
const collections_api_js_1 = require("./api/collections-api.js");
const points_api_js_1 = require("./api/points-api.js");
const service_api_js_1 = require("./api/service-api.js");
const snapshots_api_js_1 = require("./api/snapshots-api.js");
const errors_js_1 = require("./errors.js");
const openapi_typescript_fetch_1 = require("@qdrant/openapi-typescript-fetch");
function createApis(baseUrl, args) {
    const client = createClient(baseUrl, args);
    return {
        cluster: (0, cluster_api_js_1.createClusterApi)(client),
        collections: (0, collections_api_js_1.createCollectionsApi)(client),
        points: (0, points_api_js_1.createPointsApi)(client),
        service: (0, service_api_js_1.createServiceApi)(client),
        snapshots: (0, snapshots_api_js_1.createSnapshotsApi)(client),
    };
}
exports.createApis = createApis;
function createClient(baseUrl, { headers, timeout }) {
    const use = [];
    if (Number.isFinite(timeout)) {
        use.push(async (url, init, next) => {
            const controller = new AbortController();
            const id = setTimeout(() => controller.abort(), timeout);
            try {
                return await next(url, { ...init, signal: controller.signal });
            }
            catch (e) {
                if (e instanceof Error && e.name === 'AbortError') {
                    throw new errors_js_1.QdrantClientTimeoutError(e.message);
                }
                throw e;
            }
            finally {
                clearTimeout(id);
            }
        });
    }
    use.push(async (url, init, next) => {
        const response = await next(url, init);
        if (response.status === 200 || response.status === 201) {
            return response;
        }
        throw errors_js_1.QdrantClientUnexpectedResponseError.forResponse(response);
    });
    const client = openapi_typescript_fetch_1.Fetcher.for();
    client.configure({ baseUrl, init: { headers }, use });
    return client;
}
exports.createClient = createClient;
